src/seedu/addressbook/Main.java

```diff
@@ -104,12 +108,15 @@ private void recordResult(CommandResult result) {
      * @return result of the command
      */
     private CommandResult executeCommand(Command command)  {
-        try {
+    	try {
+    		
             command.setData(addressBook, lastShownList);
             CommandResult result = command.execute();
             storage.save(addressBook);
             return result;
-        } catch (Exception e) {
+        
+        }
+        catch (Exception e) {
```

Comment:
when we catch an exception, we try to be as specific as possible, because there are a lot of different exceptions, and different exceptions should have different handling method.